syntax = "proto3";

package qmtl.callback;

enum NodeCallbackType {
    ON_EXECUTE = 0;
    ON_STOP = 1;
    ON_REFCOUNT_ZERO = 2;
}

message NodeCallbackRequest {
    string node_id = 1;
    NodeCallbackType callback_type = 2;
    string url = 3;
    map<string, string> metadata = 4;
}

message NodeCallbackResponse {
    bool success = 1;
    string message = 2;
    string callback_id = 3;
}

message NodeCallbackEvent {
    string node_id = 1;
    NodeCallbackType callback_type = 2;
    map<string, string> event_payload = 3;
    int64 triggered_at = 4;
}
